### PR TITLE
[Users] Add bio character limit validation and form handling

### LINKED ISSUE #1488: [Users] User Profile Bio Character Limit & Validation
The UserProfile.bio field currently has no character limit, which could cause UI issues

### PR DESCRIPTION
**Pull request by harriet-fisher**
Addresses issue #1488
This PR implements bio character limit validation for user profiles. I added a 500-character limit with form validation and user feedback, implemented new views for bio editing for users with profiles, as well as implemented tests to validate bio length.

Backend Changes:
- users/models.py: Added max_length=500 constraint to the bio field in UserProfile model
- users/forms.py: Created new UserProfileForm with bio validation and textarea widget
- users/views.py: Added edit_profile view to handle profile editing with form validation
- users/urls.py: Added url for the edit profile functionality

Frontend Changes:
- src/templates/users/edit_profile.html: New template for editing user profiles with clean form layout
- src/templates/users/userprofile_detail.html: Added "Edit Profile" button for users viewing their own profile

Added tests for UserProfileForm including:
- Valid bio submission
- Bio character limit enforcement (500+ characters rejected)
- Edge case testing  at exactly 500 characters
- Empty bio handling

### CODE REVIEW

File: src/chigame/users/tests/test_forms.py [hunk 1]
```diff
@@ -1,36 +1,101 @@
...
+        """
+        profile = UserProfile.objects.create(user=user, bio="")
+        long_bio = "a" * 501  # 501 characters
+        form = UserProfileForm({"bio": long_bio}, instance=profile)
+
+        assert not form.is_valid()
+        assert "bio" in form.errors
+        assert "Ensure this value has at most 500 characters" in form.errors["bio"][0]
+
+    def test_bio_exactly_500_characters(self, user: User):
```
Comment (davdma) [inline src/chigame/users/tests/test_forms.py:78]: Kudos for adding tests as well for your new implementation!

File: src/chigame/users/views.py [hunk 2]
```diff
@@ -792,3 +793,31 @@ def notifications_by_label(request, label_id):
         "notifications": notifications,
     }
     return render(request, "users/notifications_by_label.html", context)
+
+
+@login_required
+def edit_profile(request, pk):
+    """
+    View for editing user profile information including bio.
```
Comment (davdma) [inline src/chigame/users/views.py:866]: This was something very much needed! We had views for editing user account info, but not profile (as I wanted to keep them separate). Being able to edit the bio was a crucially missing feature so thank you!

**Review by davdma — APPROVED** [review]
Great work Harriet on flushing out the user profile UI - the bio limit and profile editing was sorely needed, so I'm happy to see that you took the initiative.

**Review by cmarshall282 — APPROVED** [review]
Looks great!

**Comment by cmarshall282** [comment]
Issue Score: Excellent

Comments:
Great job! Here is some feedback:

- Make sure that all commits follow the format specified in the ChiGame Developer Guidelines

**Comment by cmarshall282** [comment]
Issue Score: Excellent

Comments:
Great job! Here is some feedback:

- Make sure that all commits follow the format specified in the ChiGame Developer Guidelines

**Comment by cmarshall282** [comment]
Issue Score: Excellent

Comments:
Great job! Here is some feedback:

- Make sure that all commits follow the format specified in the ChiGame Developer Guidelines
